### ISSUE TITLE
Switching between `passed` and `failed` tabs breaks UI state of the Approved/Approve button

### ISSUE THREAD

**Issue by shankarsridhar** [garris/BackstopJS#1166]
**TL;DR:**
- Display logic for `Approve` button seems to be inconsistent with the backing data for the UI.
- Filtering is not working unless page is refreshed.

After backstop tests are run, and backstop report page is opened (by running `ember backstop-report`), the `Approve` button,  if present, comes back after switching between the `passed` and `failed` tabs.

Just to be clear, here is what I am referring to as the `passed` and `failed` tabs:
![image](https://user-images.githubusercontent.com/4473214/77271852-f7ed1900-6c6c-11ea-8de5-1775ff0c4d46.png)

*Expectation:*
- The `Approve` button should not be present for tests that have already been approved.
- The test that was approved (by clicking on `Approve` button) should be filtered as passed.

*What actually happens:*
- The `Approve` button is present for tests that have already been approved.
- Filtering tests bases on passed/failed seems to be _not_ working. Tests that have been approved are still displayed in the failed category.  

If the backstop report page is refreshed, then all the UI state seems to be updated/rendered correctly as per the backing data.

Here is a screen capture to illustrate the problem:
[Approve-button-bug.mov.zip](https://github.com/garris/BackstopJS/files/4366817/Approve-button-bug.mov.zip)

**Comment by garris** [comment]
Great catch @shankarsridhar!

@1276stella is this because we are not saving `approved` state? (my guess is we simply update the DOM)

**Comment by 1276stella** [comment]
@garris a PR is opened to fix this issue. Please see the detailed explanations in the PR. https://github.com/garris/BackstopJS/pull/1175 Thank you!
